### ISSUE TITLE
Feature: when defining questions, assign arrow key positions randomly

### ISSUE THREAD

**Issue by SimplGy** [zhouzi/logicalornot#3]
Because the writer of the question must decide which answer is on the `left`, `right`, and `up` position, a bias could creep in where correct answers are assigned to a certain position more often.

Would it be better to randomly assign a cursor position to answers?

**Comment by SimplGy** [comment]
Alternative question syntax:

``` js
{
  "question": "1 == '1'",
  "answers": [ "false", "true", "undefined"] // by convention, `answers[0]` is the correct one, but randomize the display
}
```

This makes it briefer, and possibly easier, to contribute questions, and removes the chance for bias when writing the question.

**Comment by zhouzi** [comment]
Thanks for the suggestion @SimplGy!

Yes, you're right, the questions' model is way too complex and it's painful to add new ones. I like the simplicity of yours. However, I think players shouldn't have to spend too much time searching for the right button. It's probably better if the answers are ordered "logically" rather than randomly.

Maybe a gulp task could parse the simplified file and build the proper json by searching for the right answer using `eval("'foo' || 'bar'")`. What do you think?

**Comment by SimplGy** [comment]
You could do the `eval`, either at build time (or, since your app is not a perf hog, run time). It has a few downsides:
1. If the answer is a blank string, it's easier to communicate that with `(blank string)` than `''` sometimes. There are other cases that might be better explained that represented with syntax.
2. Array and object equality...
3. Other cases I'm not thinking of

I'd suggest keeping the question contribution streamlined and parse `question.answers` for when you want to make sure certain answers are always in certain positions. You could make sure that if the answers are `true/false`, they're always put on the left and right. If it's `-1,0,1`, you can enforce the layout you like, and so on. Then it's in the app layer, which is maybe the better place for it.

**Comment by zhouzi** [comment]
Thanks for the suggestions @SimplGy! I've included many of them in the last release (randomly displayed answers, simplified structure for the questions and so on). I've moved to a "dynamic" calculation of the result of the questions so you don't need to indicate which answer is correct. After some tests and updates, it seems to work properly so I'll stick with it for now.

Also, you are right about the restriction to "logical operators". I've removed it as I was already running out of questions. I'll probably include questions about some other topic in a near future, adding back the ones you submitted. If you have some others in mind, feel free to submit it :)
